polar-core/src/partial/simplify.rs

```diff
@@ -597,44 +601,55 @@ impl Simplifier {
     where
         F: Fn(&mut Self, &mut Operation, &TermSimplifier) + 'static + Clone,
     {
-        *term = self.deref(term);
-        if matches!(
-            term.value(),
-            Value::Dictionary(_) | Value::Call(_) | Value::List(_) | Value::Expression(_)
-        ) {
-            let value = term.mut_value();
-            match value {
-                Value::Dictionary(dict) => {
-                    for (_, v) in dict.fields.iter_mut() {
-                        self.simplify_term(v, simplify_operation.clone());
-                    }
+        if self.seen.contains(term) {
```

Comment:
I don't have the context to review this, what's this doing?

Reply:
see [here](https://www.notion.so/osohq/Simplifier-infinite-recursion-ad0479a286854af687b40f7e00db0876). it's kind of a simple-minded fix, I know there are still expressions that the simplifier will choke on, although I'm not sure how to produce them now. I actually think this might not be needed anymore now that we supply aliases to the simplifier, i'll do some testing